### PR TITLE
Align namespaces on internal types

### LINKED ISSUE #110: Align namespaces in the project
Currently, there are following namespaces in the project:
- [x] `NServiceBus` - containing only public API
- [x] `NServiceBus.Azure.Transports.WindowsAzureStorageQueues` - only `[ObsoleteEx]`
- [ ] `NServiceBus.AzureStorageQueues` - internal types to be moved to `NServiceBus.Transports.AzureStorageQueues`
- [ ] `NServiceBus.AzureStorageQueues.Config` - internal types to be moved to `NServiceBus.Transports.AzureStorageQueues`
- [x] `NServiceBus.Config` - only `[ObsoleteEx]`
- [x] `NServiceBus.Features` - only `[ObsoleteEx]`
- [ ] `TaskEx`  to be moved to `NServiceBus.Transports.AzureStorageQueues`

### PR DESCRIPTION
**Pull request by boblangley**
Fixes #110

### CODE REVIEW

**Comment by boblangley** [comment]
https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/b1fcc31a64343f7013f0c7797cab48131e8ae7e9/src/Transport/DelayDelivery/DelayedMessageEntity.cs#L1

@yvesgoeleven @SeanFeldman 
This type I did not adjust. I don't think it matters to change it, but since it is the TableEntity I wasn't sure if changing the namespace may cause potential migration issues for customers with existing timeouts.

If my concern isn't necessary, then feel free to adjust it and add that to the PR.

**Comment by SeanFeldman** [comment]
> since it is the TableEntity I wasn't sure if changing the namespace may cause potential migration issues for customers with existing timeouts.

We persist data and objects are treated as `TableEntity`, not as specific types such as `DelayedMessageEntity` AFAIK. `PartitionKey` and `RowKey` are the only two things that matter. @yvesgoeleven can confirm.

**Comment by SeanFeldman** [comment]
Gotta love R# inspections...

**Comment by yvesgoeleven** [comment]
@SeanFeldman correct, type doesn't matter

**Comment by SeanFeldman** [comment]
Thanks @yvesgoeleven for confirmation. In that case, 👍 for changing that namespace as well.

File: src/Transport/TaskEx.cs [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+        // Used to explicitly suppress the compiler warning about
+        // using the returned value from async operations
+        public static void Ignore(this Task task)
+        {
+        }
 
-    //TODO: remove when we update to 4.6 and can use Task.CompletedTask
-    public static readonly Task CompletedTask = Task.FromResult(0);
+        //TODO: remove when we update to 4.6 and can use Task.CompletedTask
+        public static readonly Task CompletedTask = Task.FromResult(0);
```
Comment (yvesgoeleven) [inline src/Transport/TaskEx.cs]: Shouldn't this be removed now we are 4.6?

Comment (yvesgoeleven) [reply src/Transport/TaskEx.cs]: Ow wait, asb was 4.6, this isn't right?

Comment (SeanFeldman) [reply src/Transport/TaskEx.cs]: Correct. There's no FW change here.

**Comment by yvesgoeleven** [comment]
@SeanFeldman @boblangley eventhough it doesn't matter for storage, ensure that this serialization doesn't need the type https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/develop/src/Transport/DelayDelivery/DelayedMessageEntity.cs#L20

**Comment by SeanFeldman** [comment]
> @SeanFeldman @boblangley eventhough it doesn't matter for storage, ensure that this serialization doesn't need the type https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/develop/src/Transport/DelayDelivery/DelayedMessageEntity.cs#L20

@yvesgoeleven it's a private method used internally to serialize headers, which doesn't depend on namespace of this containing class https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/83461f22c45cf7212496af4fc8e9f12d42e2caa3/src/Transport/DelayDelivery/DelayedMessageEntity.cs#L33

**Comment by SeanFeldman** [comment]
[Pushed update](https://github.com/Particular/NServiceBus.AzureStorageQueues/pull/336/commits/acb93e924ccd118d9f4e85a69a1c52d673e9a37c) to the namespace. Once green, can be reviewed and if the rest is OK, merged.

**Comment by SeanFeldman** [comment]
@Particular/azure-maintainers last review and merge (squash)?

**Comment by boblangley** [comment]
👍 to your changes @SeanFeldman

**Pull request merged**